EIPS/eip-5792.md

```diff
@@ -31,42 +31,77 @@ In updating to a new set of `wallet_` RPCs, our main goals are to enforce a clea
+The items in the `calls` field are simple `{to, data, value}` tuples.
 
 The capabilities field is how an app can communicate with a wallet about capabilities a wallet supports. For example, this is where an app can specify a paymaster service URL from which an [ERC-4337](./eip-4337.md) wallet can request a `paymasterAndData` input for a user operation.
 
+Each capability defined in the "capabilities" member can define global or call specific fields.
+These fields are set inside this capability's entry in the `capabilities` object.
+Each entity in the `calls` field may contain an optional `capabilities` object.
+This object allows the applications to attach a capability-specific metadata to individual calls.
+
+Unless these requirements are explicitly overriden by a certain `capability`, the wallet must adhere to the following rules.
+Note that such a `capability` is not in the scope of this EIP and should be defined in its own ERC.
+
 The wallet:
 
 * MUST send the calls in the order specified in the request
-* MUST send the calls on the same chain identified by the call's `chainId` property
+* MUST send the calls on the same chain identified by the request's `chainId` property
+* MUST send the calls from the address specified in the request's `from` property, if provided
+  * If `from` is not provided the wallet SHOULD present the user with an opportunity to view and change the `from` address during confirmation
 * MUST NOT await for any calls to be finalized to complete the batch
+* MUST submit multiple calls as an atomic unit in a single transaction
 * MUST NOT send any calls from the request if the user rejects the request
 * MAY revert all calls if any call fails
-* MAY send all calls as part of one or more transactions, depending on wallet capability
-* SHOULD stop executing the calls if any call fails
+* MUST not execute any further calls after a failed call
 * MAY reject the request if the from address does not match the enabled account
 * MAY reject the request if one or more calls in the batch is expected to fail, when simulated sequentially
+* MUST reject the request if it contains a `capability` (either top-level or call-level) that is not supported by the wallet and the `capability` is not explicitly marked as optional.
+  * Applications may mark a capability as optional by setting `optional` to `true`. See the RPC Specification section below for details.
+
+If provided, the wallet MUST respect the `id` field and return it in the response.
+
+Identifiers, whether provided by the app or generated by the wallet, MUST be a unique string up to 4096 bytes (8194 characters including leading `0x`).
+
+App-provided `id`s MUST be unique per sender per app, where each "app" SHOULD be identified by their domain.
+
+Wallets MUST reject requests with duplicate `id`s.
+
+Within 24 hours from the corresponding `wallet_sendCalls`, wallets SHOULD return a call-batch status when `wallet_getCallsStatus` is called with the same `id`.
```

Comment:
Does it mean that the wallet should store the `wallet_sendCalls` result for 24hours? What's the reasoning for that?

Reply:
But, if the wallets do not store and return the results of `wallet_sendCalls`, how would an app ever get them?

Reply:
My question is about the reasoning for 24 hours. Wondering if it's just an arbitrary number or sth else.
I'd even consider changing SHOULD to MUST due to having `wallet_getCallsStatus` calls guaranteed

Reply:
Ok, got it. Yes, "24 hours" is an arbitrary number, and that is why I wrote SHOULD - if the value is not available after, say, 19 hours it is probably not such a big deal. I guess in the vast majority of cases the `wallet_getCallsStatus` will be only relevant for a couple of minutes. 
We can change it to MUST if we want to be strict about it, and we can increase the time window if there is a potential use-case for it.

Reply:
it's odd grammatically, because it could be read as "wallets are expected to return a call status at some point in the next 24 hours" or it cold be read as "apps can assume they're never getting a call-batch if 24 hours pass". maybe worth breaking wallet behavior and app behavior into 2 distinct sentences?

Reply:
> "wallets are expected to return a call status at some point in the next 24 hours" or it cold be read as "apps can assume they're never getting a call-batch if 24 hours pass".

Frankly, I feel fine leaving any behaviour after the 24-hour window udefined, as this is an extremely long time for a transaction.
As for the meaning within the 24 window, clearly the wallet will want to return a status as soon as possible, and the dapps will not be able to work with a wallet that is not able to do so.
If you can suggest a more clear definition of wallet/app expectations, I would appreciate if you could make a PR.